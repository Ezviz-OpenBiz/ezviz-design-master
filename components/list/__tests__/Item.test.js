import React from 'react';
import { render, mount } from 'enzyme';
import List from '..';
import ConfigProvider from '../../config-provider';

describe('List Item Layout', () => {
  const data = [
    {
      key: 1,
      href: 'https://saastest3.ys7.com/ezd/',
      title: `EZ Design`,
      avatar: 'https://joeschmoe.io/api/v1/random',
      description:
        'EZ Design, a design language for background applications, is refined by EZVIZ Team.',
      content:
        'We supply a series of design principles, practical patterns and high quality design resources (Sketch and Axure), to help people create their product prototypes beautifully and efficiently.',
      extra: 'extra',
    },
  ];

  it('horizontal itemLayout List which contains string nodes should not be flex container', () => {
    const wrapper = mount(
      <List
        dataSource={data}
        renderItem={item => (
          <List.Item key={item.title}>
            I am <span>ezd</span> design list item
          </List.Item>
        )}
      />,
    );
    expect(wrapper.find('.ezd-list-item').at(0).hasClass('ezd-list-item-no-flex')).toBe(true);
  });

  it('horizontal itemLayout List should be flex container defaultly', () => {
    const wrapper = mount(
      <List
        dataSource={data}
        renderItem={item => (
          <List.Item key={item.title}>
            <List.Item.Meta
              title={<a href={item.href}>{item.title}</a>}
              description={item.description}
            />
          </List.Item>
        )}
      />,
    );
    expect(wrapper.find('.ezd-list-item').at(0).hasClass('ezd-list-item-no-flex')).toBe(false);
  });

  it('vertical itemLayout List should be flex container when there is extra node', () => {
    const wrapper = mount(
      <List
        itemLayout="vertical"
        dataSource={data}
        renderItem={item => (
          <List.Item key={item.title} extra={item.extra}>
            <List.Item.Meta
              title={<a href={item.href}>{item.title}</a>}
              description={item.description}
            />
          </List.Item>
        )}
      />,
    );
    expect(wrapper.find('.ezd-list-item').at(0).hasClass('ezd-list-item-no-flex')).toBe(false);
  });

  it('vertical itemLayout List should not be flex container when there is not extra node', () => {
    const wrapper = mount(
      <List
        itemLayout="vertical"
        dataSource={data}
        renderItem={item => (
          <List.Item key={item.title}>
            <List.Item.Meta
              title={<a href={item.href}>{item.title}</a>}
              description={item.description}
            />
          </List.Item>
        )}
      />,
    );
    expect(wrapper.find('.ezd-list-item').at(0).hasClass('ezd-list-item-no-flex')).toBe(true);
  });

  it('horizontal itemLayout List should accept extra node', () => {
    const wrapper = mount(
      <List
        dataSource={data}
        renderItem={item => (
          <List.Item
            key={item.title}
            actions={[<a key="action">Action</a>]}
            extra={<span>{item.extra}</span>}
          >
            <List.Item.Meta
              title={<a href={item.href}>{item.title}</a>}
              description={item.description}
            />
          </List.Item>
        )}
      />,
    );
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('should render in RTL direction', () => {
    const wrapper = mount(
      <ConfigProvider direction="rtl">
        <List
          dataSource={data}
          renderItem={item => (
            <List.Item
              key={item.title}
              actions={[<a key="action">Action</a>]}
              extra={<span>{item.extra}</span>}
            >
              <List.Item.Meta
                title={<a href={item.href}>{item.title}</a>}
                description={item.description}
              />
            </List.Item>
          )}
        />
      </ConfigProvider>,
    );
    expect(render(wrapper)).toMatchSnapshot();
  });

  it('rowKey could be string', () => {
    const dataWithId = [
      {
        id: 1,
        title: `EZ Design`,
      },
      {
        id: 2,
        title: `EZ Design`,
      },
      {
        id: 3,
        title: `EZ Design`,
      },
    ];
    const wrapper = mount(
      <List
        dataSource={dataWithId}
        rowKey="id"
        renderItem={item => <List.Item>{item.title}</List.Item>}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('rowKey could be function', () => {
    const dataWithId = [
      {
        id: 1,
        title: `EZ Design`,
      },
      {
        id: 2,
        title: `EZ Design`,
      },
      {
        id: 3,
        title: `EZ Design`,
      },
    ];
    const wrapper = mount(
      <List
        dataSource={dataWithId}
        rowKey={item => item.id}
        renderItem={item => <List.Item>{item.title}</List.Item>}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
