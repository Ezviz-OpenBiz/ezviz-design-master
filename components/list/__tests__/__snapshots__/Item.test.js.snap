// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Item Layout horizontal itemLayout List should accept extra node 1`] = `
<div
  class="ant-list ant-list-split"
>
  <div
    class="ant-spin-nested-loading"
  >
    <div
      class="ant-spin-container"
    >
      <ul
        class="ant-list-items"
      >
        <li
          class="ant-list-item"
        >
          <div
            class="ant-list-item-meta"
          >
            <div
              class="ant-list-item-meta-content"
            >
              <h4
                class="ant-list-item-meta-title"
              >
                <a
                  href="https://ant.design"
                >
                  EZ Design
                </a>
              </h4>
              <div
                class="ant-list-item-meta-description"
              >
                EZ Design, a design language for background applications, is refined by Ant UED Team.
              </div>
            </div>
          </div>
          <ul
            class="ant-list-item-action"
          >
            <li>
              <a>
                Action
              </a>
            </li>
          </ul>
          <span>
            extra
          </span>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`List Item Layout rowKey could be function 1`] = `
<List
  dataSource={
    Array [
      Object {
        "id": 1,
        "title": "EZ Design",
      },
      Object {
        "id": 2,
        "title": "EZ Design",
      },
      Object {
        "id": 3,
        "title": "EZ Design",
      },
    ]
  }
  renderItem={[Function]}
  rowKey={[Function]}
>
  <div
    className="ant-list ant-list-split"
  >
    <Spin
      size="default"
      spinning={false}
      wrapperClassName=""
    >
      <div
        className="ant-spin-nested-loading"
      >
        <div
          className="ant-spin-container"
          key="container"
        >
          <ul
            className="ant-list-items"
          >
            <Item>
              <li
                className="ant-list-item"
              >
                EZ Design
              </li>
            </Item>
            <Item>
              <li
                className="ant-list-item"
              >
                EZ Design
              </li>
            </Item>
            <Item>
              <li
                className="ant-list-item"
              >
                EZ Design
              </li>
            </Item>
          </ul>
        </div>
      </div>
    </Spin>
  </div>
</List>
`;

exports[`List Item Layout rowKey could be string 1`] = `
<List
  dataSource={
    Array [
      Object {
        "id": 1,
        "title": "EZ Design",
      },
      Object {
        "id": 2,
        "title": "EZ Design",
      },
      Object {
        "id": 3,
        "title": "EZ Design",
      },
    ]
  }
  renderItem={[Function]}
  rowKey="id"
>
  <div
    className="ant-list ant-list-split"
  >
    <Spin
      size="default"
      spinning={false}
      wrapperClassName=""
    >
      <div
        className="ant-spin-nested-loading"
      >
        <div
          className="ant-spin-container"
          key="container"
        >
          <ul
            className="ant-list-items"
          >
            <Item>
              <li
                className="ant-list-item"
              >
                EZ Design
              </li>
            </Item>
            <Item>
              <li
                className="ant-list-item"
              >
                EZ Design
              </li>
            </Item>
            <Item>
              <li
                className="ant-list-item"
              >
                EZ Design
              </li>
            </Item>
          </ul>
        </div>
      </div>
    </Spin>
  </div>
</List>
`;

exports[`List Item Layout should render in RTL direction 1`] = `
<div
  class="ant-list ant-list-split ant-list-rtl"
>
  <div
    class="ant-spin-nested-loading"
  >
    <div
      class="ant-spin-container"
    >
      <ul
        class="ant-list-items"
      >
        <li
          class="ant-list-item"
        >
          <div
            class="ant-list-item-meta"
          >
            <div
              class="ant-list-item-meta-content"
            >
              <h4
                class="ant-list-item-meta-title"
              >
                <a
                  href="https://ant.design"
                >
                  EZ Design
                </a>
              </h4>
              <div
                class="ant-list-item-meta-description"
              >
                EZ Design, a design language for background applications, is refined by Ant UED Team.
              </div>
            </div>
          </div>
          <ul
            class="ant-list-item-action"
          >
            <li>
              <a>
                Action
              </a>
            </li>
          </ul>
          <span>
            extra
          </span>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
